code/Model/SiteTree.php

```diff
@@ -951,16 +946,26 @@ public function setParent($item)
 
     /**
      * Get the parent of this page.
+     * This is here to keep compatability with the has_one magic method
+     * that always returns an object, even if there is none for the specific relation.
      *
-     * @return SiteTree Parent of this page
+     * @return SiteTree Parent of this page or SiteTree singleton
+     */
+    public function Parent()
+    {
+        return $this->getParent() ?: Injector::inst()->get(self::class);
+    }
+
+    /**
+     * Get the parent of this page, if any
+     *
+     * @return SiteTree|null Parent of this page
      */
     public function getParent()
     {
-        $parentID = $this->getField("ParentID");
-        if ($parentID) {
-            return SiteTree::get_by_id(self::class, $parentID);
+        if ($this->ParentID) {
+            return SiteTree::get_by_id(self::class, $this->ParentID);
```

Comment:
```suggestion
            return SiteTree::get()->byID($this->ParentID);
```
Slightly nicer syntax

Reply:
There is a difference between byID and get_by_id - from the code:

```php
// The object returned is cached, unlike DataObject::get()->byID() {@link DataList::byID()}
```